exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -347,6 +347,59 @@ func (ku *Kucoin) UpdateTradablePairs(ctx context.Context, forceUpdate bool) err
 	return nil
 }
 
+// GetFuturesTickers does n * REST requests based on enabled pairs of the futures asset type
+func (ku *Kucoin) GetFuturesTickers(ctx context.Context) ([]*ticker.Price, error) {
+	pairs, err := ku.GetEnabledPairs(asset.Futures)
+	if err != nil {
+		return nil, err
+	}
+
+	var wg sync.WaitGroup
+	tickersC := make(chan *ticker.Price, len(pairs))
+	errC := make(chan error, len(pairs))
+
+	for i := range pairs {
+		p, err := ku.FormatExchangeCurrency(pairs[i], asset.Futures)
+		if err != nil {
+			errC <- err
+			break
+		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			if tick, err := ku.GetFuturesTicker(ctx, p.String()); err != nil {
```

Comment:
The purpose of the wrapper function `GetTickers` is to batch request tickers. This is not doing that, and is instead going to be slow for large numbers of pairs. I can't be happy merging this for the loss of efficiency versus the existing

There is certainly an argument to be had that the commentary on the function doesn't state that, so I'm not putting any blame anywhere.

Yes by doing the batch, one can lose out on some data, for example Huobi doesn't provide `Last` in their batch endpoint, but it can be retrieved in other ways in other functions. Every other wrapper implementation of the function only makes minimal API calls to retrieve the largest amount of ticker data. For here, bid and ask data can still be retrieved via the orderbook

edit: I'm just re-summarising what Thrasher has said 😅

Reply:
noted, I have reverted UpdateTickers